packages/swapper/src/api.ts

```diff
@@ -4,9 +4,36 @@ import type { CosmosSdkChainId, EvmChainId, UtxoChainId } from '@shapeshiftoss/c
 import { createErrorClass } from '@shapeshiftoss/errors'
 import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
 import type { ChainSpecific, KnownChainIds, UtxoAccountType } from '@shapeshiftoss/types'
+import type { Result } from '@sniptt/monads'
 
 export const SwapError = createErrorClass('SwapError')
 
+export type SwapErrorMonad<T = any> = {
```

Comment:
Temporary name until the previous `SwapError` error class is removed

Reply:
`SwapError` Not going to be removed in this PR - 173 LSP references for it. 

Going to have to do with this for now in catch clauses:
```typescript
    if (e instanceof SwapError)
      return Err(
        makeSwapErrorMonad({
          message: e.message,
          code: e.code,
          details: e.details,
        }),
      )
```

Reply:
as discussed might want to call this `SwapErrorLeft` or similar, this isn't a monad at this point

Reply:
Renamed to `SwapErrorRight` a028c05023551a682060fd2bf10f818c10b7860a

Reply:
Sorry to nitpick, but traditionally `Right` is the correct result and `Left` is error